src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -96,6 +109,15 @@ public int stem(char s[], int len) {
     if (len < 3 || stopwords.contains(s, 0, len))
       return len;
 
+    String s_str = new String(s); //a String representation of the term's char[]
```

Comment:
Anywhere else I'd agree but in this code we're having similar comparisons everywhere if you take a look at the rest of the code. Since this is the case it might look a bit weird when *only* there we're using a constant and everywhere else directly the number.